### PR TITLE
Add deinterlace on VA module.

### PR DESCRIPTION
**Pull request by feiwan1**
Jira: None.
Test: Build passes on Linux and Android.

Signed-off-by: Wang,Fei <fei.w.wang@intel.com>

### CODE REVIEW

**Comment by kalyankondapally** [comment]
@XiaosongWei @xuguangxin ping

File: common/compositor/va/varenderer.cpp [hunk 1]
```diff
@@ -132,6 +138,34 @@ bool VARenderer::SetVAProcFilterColorValue(HWCColorControl mode, float value) {
     return false;
   }
 }
+
+bool VARenderer::SetVAProcFilterDeinterlaceValue(HWCDeinterlaceControl mode) {
```
Comment (xuguangxin) [inline common/compositor/va/varenderer.cpp]: this function called for every frame, so you set update_caps_ = true for every frame, then you need to create new deinterlace_ for every frame. this is not necessary.  could we just create/chang deinterlace_ when the value changed? I mean share same deinterlace_ for multiple frames

Comment (xuguangxin) [reply common/compositor/va/varenderer.cpp]: I saw the same issue on color balance too

**Comment by kalyankondapally** [comment]
This will need rebasing, also we need to figure out a way to test this outside Android now. Can we expand linux tests to do this ?

**Comment by xuguangxin** [comment]
Hi @kalyankondapally , @XiaosongWei to deliver end2end usage, we need to 
1. Ddd interlace information to gralloc, let media middleware set it
2. Expose deinterlace mode from hwc service. When we pause the video, user may turn on/off the deinterlace mode, they want to see the result instantly.
1 is more urgent since the interface need use by media middleware.

**Comment by kalyankondapally** [comment]
@xuguangxin Why gralloc, does this setting have any influence on how the buffers should be allocated ?

**Comment by xuguangxin** [comment]
I mean minigmb, media stack will set the interlace flag for intelaced frame(not all frame are interlaced). then the hwc need does deinterlace base on the flag. They are one exception if the user sets the force deinterlace mode , hwc will ignore the flag and always do deinterlace.  We need a private function or data structure because google did not define the deinterlace flag in graphic buffer.

**Comment by kalyankondapally** [comment]
hmm we dont need to do that via minigbm, we should propogate that via hwcservice.gralloc is not the place for it.

**Comment by xuguangxin** [comment]
the interlace flag is buffer based, one frame is interlaced, next frame may not interlaced.

**Comment by kalyankondapally** [comment]
hmm unless u need a change in buffer allocation, doesn't make any sense to have it through buffer manager. We need to send it via hwcservice only.

**Comment by xuguangxin** [comment]
No need change in allocation, but media middleware will set the interlace flag after we decode it. You mean we need communicate with hwc in media stack using binder? It may be too heavy. we may need set interlace flag for every frame.

**Comment by kalyankondapally** [comment]
hmm k, how about media initiates the possibility of setting this every frame and passes an fd. hwc will read from this fd, while media framework will write into it. Once done media will inform hwc that their is no need for this and we will stop polling/reading from this fd. I think we can extend this to any other thing i.e. protected content session as desired. Or HWC can create an fd and poll on it. HWC can share that fd with all clients who can write to that fd. Basically a standard server, client communication ?

**Comment by kalyankondapally** [comment]
but anyway this PR can go ahead without that. I will try to come up with a way for this.

**Comment by feiwan1** [comment]
@kalyankondapally, Will you use another PR to implement guangxin's suggestion? If so, I guess I needn't change anything for this patch right ?

**Comment by xuguangxin** [comment]
@kalyankondapally , could you share me related API? I hope we do not need extra IPC for every frame, it will hurt the performance.

**Comment by kalyankondapally** [comment]
@feiwan1  Please go ahead with this PR while we figure out how do we set per buffer configurations separately.

**Comment by kalyankondapally** [comment]
@feiwan1 You need rebasing this patch

**Comment by feiwan1** [comment]
Close this PR, I will use another PR for rebase.

**Pull request closed without merging**